### ISSUE TITLE
EditView crashes when removing the last character

### ISSUE THREAD

**Issue by lilianmoraru** [gyscos/Cursive#149]
`rustc`: `rustc 1.19.0 (0ade33941 2017-07-17)`
backend: `termion`(just installed libncursesw - it is reproducible with ncruses too)

Steps to reproduce:
1. `cargo run --no-default-features --features termion-backend --example edit`
2. Press `Space` and then `Backspace`

It should close with error code `101`, as if an `unwrap` was called(but the message is not printed).

**Comment by lilianmoraru** [comment]
Not reproducible in `0.5.1`( 0046443bab1a136f78c844e3db20f429d173fa6d ), I'll bisect

**Comment by lilianmoraru** [comment]
The issue that introduced the regression is:
```
commit 05bac7a192fecc39dd043454f1b15ef1fcebf5b3
Author: Alexandre Bury <alexandre.bury@gmail.com>
Date:   Mon Aug 14 16:32:01 2017 -0700

    More safe subtractions all around.
```

**Comment by lilianmoraru** [comment]
This assert fails: https://github.com/gyscos/Cursive/blob/master/src/views/edit_view.rs#L402

**Comment by gyscos** [comment]
Thank you very much for the detailed report!

No idea what this assert was doing here, must have been an artifact of some obscure debugging. :(

**Comment by gyscos** [comment]
I fixed the assert in 391c5af46247a3fc94b6400c97d97b2f5d6337d9, and version `0.6.4` should have the fix.
